adb/adb_commands.py

```diff
@@ -171,19 +172,41 @@ def Pull(self, device_filename, dest_file=None, timeout_ms=None):
     Returns:
       The file data if dest_file is not set.
     """
-    if isinstance(dest_file, basestring):
-      dest_file = open(dest_file, 'w')
-    elif not dest_file:
-      dest_file = cStringIO.StringIO()
-    connection = self.protocol_handler.Open(
-        self.handle, destination='sync:',
-        timeout_ms=timeout_ms)
-    self.filesync_handler.Pull(connection, device_filename, dest_file)
-    connection.Close()
-    # An empty call to cStringIO.StringIO returns an instance of
-    # cStringIO.OutputType.
-    if isinstance(dest_file, cStringIO.OutputType):
-      return dest_file.getvalue()
+
+    if dest_file:
+      print("pull: %s -> %s" % (device_filename, dest_file))
```

Comment:
No printing, please. Also, instead of an assert on 182, raise a custom error or something like ValueError that can be caught more specifically than the generic AssertionError.